In [1]:
import numpy as np
import matplotlib.pyplot as plt
import astropy.units as u
import astropy.constants as const


In [14]:
# 3
a = 20.5 * u.AU
P = np.sqrt(a**3 * 4 * np.pi **2 / (const.G*const.M_sun))
P = P.to(u.yr)

V_esc = 11.2 * u.km / u.s
V_inf_sqr = const.G * const.M_sun * ((2/(1*u.AU))-(1/(20.5 * u.AU)))
V_imp_sqr = V_inf_sqr + V_esc**2
V_imp = np.sqrt(V_imp_sqr)
V_imp = V_imp.to(u.km/u.s)
V_inf = np.sqrt(V_inf_sqr)
V_inf = V_inf.to(u.km/u.s)
V_inf

<Quantity 41.610847673977396 km / s>

In [16]:
P = 90  * 60 * u.s
P

<Quantity 5400.0 s>

In [19]:
F_sun = (const.R_sun ** 3) / (1 * u.AU)**2
F_moon = ((1737 * u.km)**3) / (384400 * u.km)**2 
F_sun.to(u.km), F_moon

(<Quantity 15.033342959439427 km>, <Quantity 0.03546767312952277 km>)

In [20]:
P = 8.5 * u.yr
a = P ** (2/3)
a

<Quantity 4.164977092751979 yr(2/3)>

In [23]:
r = const.G * const.M_sun / (50 * u.km/u.s)**2
r = r.to(u.AU)
r

<Quantity 0.3549498420501248 AU>

P = 8.5 * u .yr
b = 12.5* u.m/u.s * P / (2 * np.pi)
b = b.to(u.AU)
M_p = (b / (4.16 * u.AU))*const.M_sun
M_p = M_p / const.M_jup
M_p,b

In [41]:
mass_b = 2 * 27 * (27e-5) + 4 * 16 * (1e-2) + 1 * 40 * (40e-5)
mass_c = 24 * (24e-4) + 3*16*(1e-2) + 56 * (56e-4)
mass_c

0.8512

In [35]:
6.3e-4 * 16

0.01008

In [40]:
(10**7)/(10**10.2)

0.0006309573444801943